lib/canary.ts

```diff
@@ -3,72 +3,38 @@ import cbuild = require('@aws-cdk/aws-codebuild');
 import events = require('@aws-cdk/aws-events');
 import cdk = require('@aws-cdk/cdk');
 
-import { Shellable, ShellPlatform } from './shellable';
+import { Shellable, ShellableProps } from './shellable';
 
-export interface CanaryProps {
+export interface CanaryProps extends ShellableProps {
```

Comment:
The reason I didn't extend `ShellableProps` is because a canary always has no source, where `Shellable` lets you configure it.